debugview/plugin.cpp

```diff
@@ -220,40 +214,24 @@ class debugview : public threadloop {
     bool load_camera_images() {
         RAC_ERRNO_MSG("debugview at start of load_camera_images");
 
-        if (last_datum_with_images == nullptr) {
+        cam = _m_cam.size() == 0 ? nullptr : _m_cam.dequeue();
+        if (cam == nullptr) {
             return false;
         }
 
-        if (last_datum_with_images->img0.has_value()) {
-            glBindTexture(GL_TEXTURE_2D, camera_textures[0]);
-            cv::Mat img0{last_datum_with_images->img0.value().clone()};
-            glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, img0.cols, img0.rows, 0, GL_RED, GL_UNSIGNED_BYTE, img0.ptr());
-            camera_texture_sizes[0] = Eigen::Vector2i(img0.cols, img0.rows);
-            GLint swizzleMask[]     = {GL_RED, GL_RED, GL_RED, GL_RED};
-            glTexParameteriv(GL_TEXTURE_2D, GL_TEXTURE_SWIZZLE_RGBA, swizzleMask);
-        } else {
-            std::cerr << "img0 has no value!" << std::endl;
-            glBindTexture(GL_TEXTURE_2D, camera_textures[0]);
-            glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, TEST_PATTERN_WIDTH, TEST_PATTERN_HEIGHT, 0, GL_RED, GL_UNSIGNED_BYTE,
-                         &(test_pattern[0][0]));
-            glFlush();
-            camera_texture_sizes[0] = Eigen::Vector2i(TEST_PATTERN_WIDTH, TEST_PATTERN_HEIGHT);
-        }
+        glBindTexture(GL_TEXTURE_2D, camera_textures[0]);
+        cv::Mat img0{cam->img0.clone()};
+        glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, img0.cols, img0.rows, 0, GL_RED, GL_UNSIGNED_BYTE, img0.ptr());
+        camera_texture_sizes[0] = Eigen::Vector2i(img0.cols, img0.rows);
+        GLint swizzleMask[]     = {GL_RED, GL_RED, GL_RED, GL_RED};
```

Comment:
Why all red?

Reply:
Because the camera images are gray-scale i.e. single channel. When converting it into an RGB image, each channel will just take the R channel value.